include/robot_interfaces/robot_logger.hpp

```diff
@@ -233,16 +234,22 @@ class RobotLogger
         output_file_.close();
     }
 
-    void append_field_data_to_robot_data(std::vector<std::vector<double>> field_data)
+    /*
+    * append_field_data_to_robot_data() appends the data corresponding to
+    * every field at the same time index to the log file.
+    */
+
+    void append_field_data_to_file(std::vector<std::vector<double>> field_data)
     {
 
       infix_ostream_iterator<double> double_iterator(output_file_, ", ");
 
       for (auto data : field_data)
       {
           std::copy(data.begin(),
-                    data.end(),
+                    data.end() - 1,
```

Comment:
What is the reason for the `- 1`?

Reply:
Some fix I was trying for the comma problem, please ignore.